addons/mrp/models/stock_move.py

```diff
@@ -85,6 +85,11 @@ def write(self, vals):
                 move_line._log_message(production, move_line, 'mrp.track_production_move_template', vals)
         return super(StockMoveLine, self).write(vals)
 
+    def _get_aggregated_properties(self, move_line=False, move=False):
+        aggregated_properties = super()._get_aggregated_properties(move_line, move)
+        aggregated_properties['bom'] = aggregated_properties['move'].bom_line_id.bom_id or False
```

Comment:
You forgot to add the BoM to the `line_key` to avoid to aggregated same product from different BoM
```suggestion
        bom = aggregated_properties['move'].bom_line_id.bom_id
        aggregated_properties['bom'] = bom or False
        aggregated_properties['line_key'] += f'_{bom.id if bom else ""}'
```